### ISSUE TITLE
Case where process exits before we can write to stdin...

### ISSUE THREAD

**Issue by byllyfish** [byllyfish/shellous#45]
```
tests/test_shellous.py::test_broken_pipe 
-------------------------------- live log setup --------------------------------
1630453746.249 DEBUG Using selector: KqueueSelector
1630453746.249 DEBUG Using selector: KqueueSelector
1630453746.250 DEBUG Close <_UnixSelectorEventLoop running=False closed=False debug=True>
-------------------------------- live log call ---------------------------------
1630453746.255 INFO Runner.__aenter__ stepin <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=None> ('macOS-10.15.7 CPython 3.10.0rc1 asyncio.unix_events._UnixSelectorEventLoop ThreadedChildWatcher')
1630453746.255 INFO Runner._start stepin <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=None>
1630453746.256 DEBUG execute program '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.10/bin/python' stdin=<pipe> stdout=<pipe> stderr=<Redirect.DEVNULL: -3>
1630453746.262 DEBUG process '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.10/bin/python' created: pid 8727
1630453746.294 DEBUG process 8727 exited with returncode 0
1630453746.296 INFO <_UnixSubprocessTransport pid=8727 running> exited with return code 0
1630453746.296 DEBUG Write pipe 12 connected: (<_UnixWritePipeTransport fd=12 idle bufsize=0>, <WriteSubprocessPipeProto fd=0 pipe=<_UnixWritePipeTransport fd=12 idle bufsize=0>>)
1630453746.297 INFO <_UnixWritePipeTransport fd=12 idle bufsize=0> was closed by peer
1630453746.298 DEBUG Read pipe 14 connected: (<_UnixReadPipeTransport fd=14 polling>, <ReadSubprocessPipeProto fd=1 pipe=<_UnixReadPipeTransport fd=14 polling>>)
1630453746.299 INFO <_UnixReadPipeTransport fd=14 polling> was closed by peer
1630453746.300 INFO execute program '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.10/bin/python': <_UnixSubprocessTransport pid=8727 returncode=0 stdin=<_UnixWritePipeTransport closed fd=12 closed> stdout=<_UnixReadPipeTransport closing fd=14 idle>>
1630453746.301 INFO Runner._start stepout <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0> ex=None
1630453746.301 INFO Runner.__aenter__ stepout <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0> ex=None
1630453746.302 INFO write_stream stepin
1630453746.303 INFO write_stream stepout ex=None
1630453746.303 INFO Runner.__aexit__ stepin <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0> exc_value=None
1630453746.303 INFO Runner._finish stepin <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0>
1630453746.303 INFO Runner._wait stepin <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0>
1630453746.304 INFO Runner._wait stepout <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0> ex=None
1630453746.305 INFO Runner._close stepin <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0>
1630453746.307 INFO Runner._close stepout <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0> ex=None
1630453746.307 INFO Runner._finish stepout <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0> ex=None
1630453746.307 INFO Runner.__aexit__ stepout <Runner '...lous-fzoxJzh3-py3.10/bin/python' pid=8727 exit_code=0> ex=None exc_value=None
FAILED
```

We might want to proactively raise an exception similar to BrokenPipeError.

**Comment by byllyfish** [comment]
Latest version of the log.  May need to synthesize a BrokenPipeError in this case...

```
------------------------------ Captured log call -------------------------------
INFO     shellous:log.py:47 Runner.__aenter__ stepin <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=None> (macOS-10.15.7 CPython 3.9.6 asyncio.unix_events._UnixSelectorEventLoop MainThread ThreadedChildWatcher)
INFO     shellous:log.py:55 Runner._start stepin <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=None>
INFO     asyncio:base_subprocess.py:211 <_UnixSubprocessTransport pid=5288 running stdin=<_UnixWritePipeTransport fd=10 idle bufsize=0> stdout=<_UnixReadPipeTransport fd=11 polling>> exited with return code 0
INFO     asyncio:unix_events.py:642 <_UnixWritePipeTransport fd=10 idle bufsize=0> was closed by peer
INFO     asyncio:unix_events.py:503 <_UnixReadPipeTransport fd=11 polling> was closed by peer
INFO     asyncio:base_events.py:1665 execute program '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.9/bin/python': <_UnixSubprocessTransport pid=5288 returncode=0 stdin=<_UnixWritePipeTransport closing fd=10 idle bufsize=0> stdout=<_UnixReadPipeTransport closing fd=11 idle>>
INFO     shellous:log.py:60 Runner._start stepout <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0> ex=None
INFO     shellous:log.py:60 Runner.__aenter__ stepout <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0> ex=None
INFO     shellous:log.py:73 copy_bytearray stepin
INFO     shellous:log.py:73 write_stream stepin
INFO     shellous:log.py:55 Runner._waiter stepin <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0>
INFO     shellous:log.py:60 Runner._waiter stepout <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0> ex=None
INFO     shellous:log.py:77 write_stream stepout ex=None
INFO     shellous:log.py:77 copy_bytearray stepout ex=None
INFO     shellous:log.py:55 Runner.__aexit__ stepin <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0> exc_value=None
INFO     shellous:log.py:55 Runner._finish stepin <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0>
INFO     shellous:log.py:55 Runner._wait stepin <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0>
INFO     shellous:log.py:60 Runner._wait stepout <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0> ex=None
INFO     shellous:log.py:55 Runner._close stepin <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0>
INFO     shellous:log.py:60 Runner._close stepout <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0> ex=None
INFO     shellous:log.py:60 Runner._finish stepout <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0> ex=None
INFO     shellous:log.py:60 Runner.__aexit__ stepout <Runner '...llous-fzoxJzh3-py3.9/bin/python' pid=5288 exit_code=0> ex=None exc_value=None

```

**Comment by byllyfish** [comment]
Fixed in 0.5.0
